### PR TITLE
dev branch: add health check endpoint

### PR DESCRIPTION
**Pull request by ericchiang**
cc @dghubble @squat @kans

### CODE REVIEW

File: server/handlers.go [hunk 1]
```diff
@@ -20,6 +20,36 @@ import (
...
+		}
+		if err := s.storage.DeleteAuthRequest(a.ID); err != nil {
+			return fmt.Errorf("delete auth request: %v", err)
+		}
+		return nil
+	}()
+
+	t := s.now().Sub(start)
+	if err != nil {
+		log.Printf("storage health check failed: %v", err)
```
Comment (squat) [inline server/handlers.go]: Nit: consistent casing for log statements.

File: server/handlers.go [hunk 2]
```diff
@@ -20,6 +20,36 @@ import (
...
 
+func (s *Server) handleHealth(w http.ResponseWriter, r *http.Request) {
+	start := s.now()
+	err := func() error {
+		// Instead of trying to introspect health, just try to use the underlying storage.
+		a := storage.AuthRequest{
+			ID:       storage.NewID(),
+			ClientID: storage.NewID(),
+
+			// Set a short expiry so if the delete fails this will be cleaned up quickly by garbage collection.
```
Comment (dghubble) [inline server/handlers.go]: If I poll this every 30 seconds and the delete fails, what's the expected behavior on the next create? Basically, does the kubernetes manifest health check need to know about this detail?

Comment (ericchiang) [inline server/handlers.go]: storage.NewID() always returns a random ID. Each call deals with a unique object so they should never impact one another.

File: server/handlers.go (hunk 1 above)
Comment (ericchiang) [inline server/handlers.go]: thanks, updated

**Pull request merged**